backend/src/blocked/blocked.service.ts

```diff
@@ -42,21 +50,27 @@ export class BlockedService {
     return new SuccessResponseDto('유저를 차단하였습니다.');
   }
 
+  async deleteBlockedUser(myId: number, userId: number): Promise<SuccessResponseDto> {
+    if (myId === userId) {
+      throw new ConflictException('본인 자신은 차단되어 있지 않습니다!!!');
+    }
+    await this.userService.findExistUserById(userId);
+    const blockedUser = await this.blockedUserRepository.findOneBy({ userId: myId, blockedUserId: userId });
+    if (blockedUser === null) {
+      throw new NotFoundException('차단한 적이 없는 유저입니다.');
+    }
+    await this.blockedUserRepository.delete(blockedUser);
```

Comment:
앗 delete 에 `blockedUser` 엔티티를 그냥 바로 넣어도 되나요?

`friend` API 로 실험 결과 Entity 를 그냥 삽입하면 where 절에 entity 의 모든 값이 들어가는 군요.. id 만 전달하는 방향으로 수정해주세요! where 절에 조건 하나만 있는 편이 훨씬 빠를 것 같습니다.

1. entity 를 그대로 넘겼을 때 -> 전체 조건으로 인식되어서 where 조건 여러개로 쿼리 실행
<img width="525" alt="Screenshot 2023-04-20 at 10 27 44 PM" src="https://user-images.githubusercontent.com/71019735/233381062-41920295-2f80-4229-bb3f-60f4f5556261.png">

2. id 만 넘길 경우 -> 인덱싱된 id 로 찾아서 삭제
<img width="508" alt="Screenshot 2023-04-20 at 10 30 41 PM" src="https://user-images.githubusercontent.com/71019735/233381854-ca6df427-c6ec-4f27-9731-f59e5d6f27cf.png">

```suggestion
    await this.blockedUserRepository.delete(blockedUser.id);
```

Reply:
아 근데 blocked user 테이블이 컬럼이 userid와 blockedUserId 밖에 없고 두컬럼이 복합으로 pk라서 그냥 이렇게 delete하는게 pk 쓰는 것과 같습니다. 그리고 이렇게 해야 에러 없이 delete 할 수 있어요! 복합키라서 그런듯 합니다.

Reply:
아하 그렇네요!! 그렇다면 db 구조가 변경되지 않는 이상 괜찮을...것 같긴 하지만 entity에는 다른 프로퍼티들이 추가될 가능성도 있어서 id 랑blockedUserId로 criteria 를 명시해주셨으면 합니다!